.github/workflows/auto-deployments.yml

```
@@ -0,0 +1,97 @@
+        run: echo "matrix={\"container\":${{ steps.changed-files.outputs.all_changed_files }}}" >> "$GITHUB_OUTPUT"
+
+  matrix-job:
+    name: Job for ${{ matrix.container }}
+    runs-on: self-hosted
+    needs: [get-changes-for-spas]
+    strategy:
+      matrix: ${{ fromJSON(needs.get-changes-for-spas.outputs.matrix) }}
+      max-parallel: 4
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        if: contains(matrix.container, '-spa')
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v3
+        if: contains(matrix.container, '-spa')
+        with:
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: 'packages/**/package-lock.json'
+      - name: Get SPA NAME as strings
+        id: get-spa-name
+        if: contains(matrix.container, '-spa')
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const spaName = "${{ matrix.container }}".split('/')[1];
+            core.setOutput('spa-name', spaName)
+      - name: Build and deploy SPA on DEV
+        if: contains(matrix.container, '-spa')
+        env:
+          SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}dev"
+          GITLAB_ENV_URL: "${{ secrets.GITLAB_URL }}spa-configs"
+        # Steps in the following script
+        # #1 cd into dir
+        # #2 install the required node_modules
+        # #3 get env file from gitlab (note: without the full url curl does not resolve the endpoint)
+        # #4 build with the .env file
+        # #5 deploy using spaship cli
+        run: |
+          cd ${{ matrix.container }}
+          npm install
+          export GITLAB_URL="$GITLAB_ENV_URL"
+          echo ${{ secrets.GITLAB_URL }}
+          echo $GITLAB_URL
+          curl -k --header "PRIVATE-TOKEN:${{ secrets.GITLAB_TOKEN }}" "$GITLAB_URL%2Fdev%2F${{ steps.get-spa-name.outputs.spa-name }}%2F.env/raw?ref=master" -o .env
+          npm run build
+          export VERSION=\$(node -p "require('./package.json').version")
+          npx @spaship/cli@latest deploy --apikey=${{ secrets.SPASHIP_API_KEY }} \
+          --env=$SPASHIP_DEPLOY_URL \
+          --ref=\$VERSION
+      # - name: Build and deploy SPA on QA
+      #   if: contains(matrix.container, '-spa')
+      #   env:
+      #     SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}qa"
+      #   run: |
+      #     cd ${{ matrix.container }}
+      #     npm install
+      #     curl -k --header "PRIVATE-TOKEN:${{ secrets.GITLAB_TOKEN }}" ""https://gitlab.cee.redhat.com/api/v4/projects/9092/repository/files/spa-configs%2Fqa%2F${{ steps.get-spa-name.outputs.spa-name }}%2F.env/raw?ref=master" -o .env
+      #     npm run build
+      #     export VERSION=\$(node -p "require('./package.json').version")
+      #     npx @spaship/cli@latest deploy --apikey=${{ secrets.SPASHIP_API_KEY }} \
+      #     --env=$SPASHIP_DEPLOY_URL \
+      #     --ref=\$VERSION
```

```suggestion
      # - name: Build and deploy SPA on QA
      #   if: contains(matrix.container, '-spa')
      #   env:
      #     SPASHIP_DEPLOY_URL: "${{ secrets.SPASHIP_DEPLOY_URL }}qa"
      #   run: |
      #     cd ${{ matrix.container }}
      #     npm install
      #     curl -k --header "PRIVATE-TOKEN:${{ secrets.GITLAB_TOKEN }}" ""https://gitlab.cee.redhat.com/api/v4/projects/9092/repository/files/spa-configs%2Fqa%2F${{ steps.get-spa-name.outputs.spa-name }}%2F.env/raw?ref=master" -o .env
      #     npm run build
      #     export VERSION=\$(node -p "require('./package.json').version")
      #     npx @spaship/cli@latest deploy --apikey=${{ secrets.SPASHIP_API_KEY }} \
      #     --env=$SPASHIP_DEPLOY_URL \
      #     --ref=\$VERSION
```

remove the commented code if you are making it back active use ```GITLAB_URL``` as the env variable

---

@riginoommen this secret isn't working

---

@riginoommen change added